executor/receiver.py

```diff
@@ -0,0 +1,60 @@
+import redis
+import json
+import logging
+
+logging.getLogger().setLevel(logging.INFO)
+logger = logging.getLogger()
+
+
+class Receiver:
+    def __init__(self, cache_prefix, pid, redis_host='localhost', redis_port=6379):
+        self.tag = '[RECIEVER]'
+        self.todo_list = f'{cache_prefix}:todo'
+        self.inprogress_list = f'{cache_prefix}:in_progress'
+        self.done_list = f'{cache_prefix}:done'
+        self.jobs_hash_name = cache_prefix
+        self.pid = str(pid)
+
+        self.initializeConnection(redis_host, redis_port)
+
+    def initializeConnection(self, redis_host, redis_port):
+        logger.info(f'{self.tag} intializing connection with redis on host: {redis_host}, port: {redis_port}')
+        try:
+            self.redis_instance = redis.Redis(host=redis_host, port=redis_port)
+            logger.info(f'{self.tag} redis instance established')
+        except Exception as err:
+            logger.exception(f'{self.tag} failed to intialize connection with redis due to: {err}')
+            raise
+
+    def get_new_job(self):
+        logger.info(f'{self.tag} checking todo list for tasks ...')
+        atomic_pipeline = self.redis_instance.pipeline()
+
+        job_meta_string = self.redis_instance.brpoplpush(self.todo_list, self.inprogress_list, timeout=0)
+        atomic_pipeline.hset(self.jobs_hash_name, self.pid, job_meta_string)
+        atomic_pipeline.execute()
+        logger.info(f'{self.tag} got new job {job_meta_string}, processing job ..')
+
+        try:
+            job_meta = json.loads(job_meta_string)
+            logger.info(f'{self.tag} job processed successfully')
+            return job_meta
+        except Exception as err:
+            logger.exception(f'{self.tag} failed to process job meta due to: {err}')
+            return None
+
+    def mark_job_as_done(self, job_meta):
+        logger.info(f'{self.tag} moving job from in progress to done list ...')
+        try:
+            job_meta_string = json.dumps(job_meta)
+
+            atomic_pipeline = self.redis_instance.pipeline()
+
+            atomic_pipeline.lrem(self.inprogress_list, 0, job_meta_string)
+            atomic_pipeline.rpush(self.done_list, job_meta_string)
+
+            atomic_pipeline.execute()
+            logger.info(f'{self.tag} job moved to done list successfully')
+            return True
+        except Exception as err:
+            logger.exception(f'{self.tag} Failed to mark job as done due to {err}')
```

Comment:
You don't do that. You just send the completed healthcheck and I move it to done.